lib/config/db/couchdb-get-vendor.js

```diff
@@ -1,30 +1,37 @@
-module.exports = checkVendor
+module.exports = getVendor
 
-var findKey = require('lodash').findKey
+var defaults = require('lodash').defaults
 
 var toCouchDbUrl = require('../../utils/pouchdb-options-to-couchdb-url')
 var removeAuth = require('../../utils/remove-auth-from-url')
 
-function checkVendor (state, couch, callback) {
+function getVendor (state, couch, callback) {
   couch({url: '/'}, function (error, response, data) {
     if (error || (response && response.statusCode !== 200)) {
       var url = toCouchDbUrl(state.db.options)
       return callback(new Error('Could not find CouchDB at ' + removeAuth(url)))
     }
 
-    var vendor = findKey(data, function (property) {
-      return /^welcome/i.test(property)
+    var vendor = defaults(data.vendor, {
+      type: getVendorType(data.vendor)
     })
 
-    if (vendor !== 'couchdb') {
+    if (vendor.type  !== 'couchdb') {
       state.server.log(
         ['database', 'warn'],
         'You are not running an official CouchDB distribution, ' +
-        'but "' + vendor + '". ' +
+        'but "' + vendor.name + '". ' +
         'This might not be fully supported. Proceed at your own risk.'
       )
     }
 
-    callback(null)
+    callback(vendor)
   })
 }
+
+
+function getVendorType (vendor) {
+  if(vendor.name === 'The Apache Software Foundation') return 'couchdb'
```

Comment:
@gr2m Push up a different approach - more along the lines of what you were thinking :)

Basically, we get/set the values if the database allows us to, and log a warning if hoodie couldn't access this information.

Providing hoodie knows where the user database is, it seems to work well with cloudant - but it seems the `secret` and `admins` properties may not be needed?